samcli/local/lambda_service/service.py

```diff
@@ -0,0 +1,128 @@
+
+    def create(self):
+        """
+        Creates a Flask Application that can be started.
+        """
+        self._app = Flask(__name__)
+
+        for function_name in self.function_name_list:
+            path = '/2015-03-31/functions/{}/invocations'.format(function_name)
+            self._app.add_url_rule(path,
+                                   endpoint=path,
+                                   view_func=self._invoke_request_handler,
+                                   methods=['POST'],
+                                   provide_automatic_options=False)
+
+        self._construct_error_handling()
+
+    def run(self):
+        """
+        This starts up the (threaded) Local Server.
+        Note: This is a **blocking call**
+
+        Raises
+        ------
+        RuntimeError
+            if the service was not created
+        """
+        if not self._app:
+            raise RuntimeError("The application must be created before running")
+
+        # Flask can operate as a single threaded server (which is default) and a multi-threaded server which is
+        # more for development. When the Lambda container is going to be debugged, then it does not make sense
+        # to turn on multi-threading because customers can realistically attach only one container at a time to
+        # the debugger. Keeping this single threaded also enables the Lambda Runner to handle Ctrl+C in order to
+        # kill the container gracefully (Ctrl+C can be handled only by the main thread)
+        multi_threaded = not self.lambda_runner.is_debugging()
+
+        LOG.debug("Local Lambda Server starting up. Multi-threading = %s", multi_threaded)
+
+        # This environ signifies we are running a main function for Flask. This is true, since we are using it within
+        # our cli and not on a production server.
+        os.environ['WERKZEUG_RUN_MAIN'] = 'true'
+
+        self._app.run(threaded=multi_threaded, host=self.host, port=self.port)
+
+    def _construct_error_handling(self):
+        """
+        Updates the Flask app with Error Handlers for different Error Codes
+
+        """
+        pass
+
+    def _invoke_request_handler(self):
+        """
+        Request Handler for the Local Lambda Invoke path. This method is responsible for understanding the incoming
+        request and invoking the Local Lambda Function
+
+        Returns
+        -------
+        A Flask Response response object as if it was returned from Lambda
+
+        """
+        flask_request = request
+
+        function_name_regex = re.compile(r'/2015-03-31/functions/(.*)/invocations')
+
+        regex_match = function_name_regex.match(flask_request.path)
+
+        function_name = ""
+
+        if regex_match:
+            function_name = regex_match.group(1)
+
+        stdout_stream = io.BytesIO()
+
+        try:
+            self.lambda_runner.invoke(function_name, {}, stdout=stdout_stream, stderr=self.stderr)
```

Comment:
why is the `event` parameter empty dict? We should be passing the request's body to event

Reply:
Because I didn't get that far just yet :). I focused on just getting the main details working. It will be apart of the next commit.